Cython/Utility/CythonFunction.c

```diff
@@ -362,6 +367,34 @@ __Pyx_CyFunction_get_annotations(__pyx_CyFunctionObject *op, CYTHON_UNUSED void
     return result;
 }
 
+static PyObject *
+__Pyx_CyFunction_get_is_coroutine(__pyx_CyFunctionObject *op, CYTHON_UNUSED void *context) {
+#if PY_VERSION_HEX >= 0x03050000
+    // on v3.5.3rc1 the marker object was introduced
+    if (op->func_is_coroutine)
+        return __Pyx_NewRef(op->func_is_coroutine);
+    if (op->flags & __Pyx_CYFUNCTION_COROUTINE) {
+        PyObject *module, *fromlist, *marker = PYIDENT("_is_coroutine");
+        fromlist = PyList_New(1);
+        if (unlikely(!fromlist)) return NULL;
+        PyList_SET_ITEM(fromlist, 0, marker);
+        module = PyImport_ImportModuleLevelObject(PYIDENT("asyncio.coroutines"), NULL, NULL, fromlist, 0);
+        Py_DECREF(fromlist);
+        if (unlikely(!module)) goto ignore;
+        op->func_is_coroutine = __Pyx_PyObject_GetAttrStr(module, marker);
+        Py_DECREF(module);
+        if (unlikely(!op->func_is_coroutine)) goto ignore;
+        return __Pyx_NewRef(op->func_is_coroutine);
+    }
+ignore:
+    PyErr_Clear();
+    op->func_is_coroutine = __Pyx_NewRef(Py_None);
+    return __Pyx_NewRef(op->func_is_coroutine);
+#else
+    return __Pyx_PyBool_FromLong(op->flags & __Pyx_CYFUNCTION_COROUTINE);
+#endif
```

Comment:
Wouldn't these fallbacks be more helpful?
```suggestion
ignore:
    PyErr_Clear();
#endif
    return __Pyx_PyBool_FromLong(op->flags & __Pyx_CYFUNCTION_COROUTINE);
```

Reply:
Checking both the pre 3.5.3rc1 version and the current one, it should work fine with False instead of None. The only disadvantage I see is that we're not caching the None for non-coroutine functions, and running the flag comparisson twice. Let me see what I can do here.

Reply:
> we're not caching the None

Ah, but then just cache the return value, whatever it is.